scripts/bot-deployment/DeployBotGCP.sh

```diff
@@ -1,20 +1,48 @@
 #!/bin/bash
 set -e
 
-if [ $# -ne 3 ]; then
-    echo "Incorrect number of arguments supplied! First argument is bot's name. Second argument is path to config file. Third is the service account the bot should run as."
-    echo "example: ./DeployBotGCP.sh ethbtc-monitor-bot ./ethbtc-monitor-bot-env.txt ethbtc-account@project-name.iam.gserviceaccount.com"
+if [ $# -ne 1 ]; then
+    echo "Incorrect number of arguments supplied! First and only argument is bot's name. From this the bot's config and service account will be inferred."
+    echo "example: ./DeployBotGCP.sh ethbtc-monitor-bot"
     exit 1
 fi
 
-echo "Deploying" $1
-echo "Using ENV file:" $2
-echo "Running as Service Account:" $3
+echo "🔥Starting deployment script for bot" $1
+
+# Bot names are <identifer>-<network>-<bot-type>. EG: ethbtc-liquidator-mainnet.
+# This cut removes the network as this is not included in a service account. ethbtc-liquidator-mainnet becomes ethbtc-liquidator
+paramToServiceAccount=$(echo $1 | cut -d'-' -f 1,3)
+serviceAccountEmail=""
+
+# Loop through all service accounts in GCP and validate that the provided bot name matches to a service account email.
+for serviceAccount in $(gcloud iam service-accounts list --format="value(EMAIL)"); do
+    if [ $paramToServiceAccount == $(echo $serviceAccount | cut -d'@' -f 1) ]; then
+        serviceAccountEmail=$serviceAccount
+    fi
+done
+
+if [ $serviceAccountEmail == "" ]; then
+    echo "Bot name provided does not match to a service account."
+    exit 1
+fi
+
+echo "Bot service account found @ " $serviceAccountEmail
+echo "Pulling bot config from GCP bucket"
+
+gsutil cp gs://bot-configs/$1.env ~/.tempUMAConfig
```

Comment:
You might consider using `mktemp` here to create the temporary file -- this has a few advantages:
1. You don't have to worry about cleaning it up. Temp directories are generally cleaned up by the system after a few days.
2. In the event of a failure, this file won't pollute the user's working directory.
3. Since you don't need to delete the file, you can print the name, and the user can open that file after the script finishes to debug the environment variables they passed in.

Reply:
@mrice32 I've updated the implementation. let me know what you think :)